### PR TITLE
Update all links that have destinations outside of /docs/grafana/latest/datasources/ to use `docs/reference` shortcode

### PR DESCRIPTION
**Pull request by jdbaldry**
Use of this shortcode means that links can resolve different depending on where the page is published.
At the moment, most links are configured to go back to Grafana documentation even if there might be a more appropriate Grafana Cloud destination.

All links have been replaced with `/docs/reference` style links. The link destination for any link label is determined when the page is built and checked with relrefs. There are no broken relrefs in the datasources docs when running `make docs` so I believe all the links are working as intended when it comes to Grafana docs.

Note that the Grafana Cloud content links to the "next" version of Grafana documentation. The `<GRAFANA VERSION>` variable is replaced with `next` in Grafana Cloud pages because we set `grafana_version: next` in the Grafana Cloud documentation front matter.
For the pages published in Grafana, `<GRAFANA VERSION>` is automatically replaced with the current version.

Additional changes:
1. When I saw a relref containing only an anchor, I replaced that with a raw Markdown anchor link. It saves some keypresses and the behavior is the same as Hugo does not check anchors exist during the build.

https://github.com/grafana/writers-toolkit/pull/282 documents the shortcode behavior in more detail.

Signed-off-by: Jack Baldry <jack.baldry@grafana.com>

### CODE REVIEW

**Review by chri2547 — APPROVED** [review]

**Review by lwandz13 — APPROVED** [review]
please wait for Loki PR https://github.com/grafana/grafana/pull/72041 to merge before merging this one - thank you!

**Comment by jdbaldry** [comment]
> 

I'll also wait for the successful backport of that PR before I merge this.

**Pull request merged**
